rust_src/src/fns.rs

```diff
@@ -306,4 +308,39 @@ pub extern "C" fn internal_equal_misc(
     }
 }
 
+/// Return list of 1 minute, 5 minute and 15 minute load averages.
+///
+/// Each of the three load averages is multiplied by 100, then converted
+/// to integer.
+///
+/// When USE-FLOATS is non-nil, floats will be used instead of integers.
+/// These floats are not multiplied by 100.
+///
+/// If the 5-minute or 15-minute load averages are not available, return a
+/// shortened list, containing only those averages which are available.
+///
+/// An error is thrown if the load average can't be obtained.  In some
+/// cases making it work would require Emacs being installed setuid or
+/// setgid so that it can read kernel information, and that usually isn't
+/// advisable.
+#[lisp_fn(min = "0")]
+pub fn load_average(use_floats: LispObject) -> LispObject {
```

Comment:
```suggestion
pub fn load_average(use_floats: bool) -> LispObject {
```

Can `use_floats` be type 'bool`?

Reply:
It looks like it can be. I just tested it out and we do get `true` for anything non-`nil` which is the behaviour I was trying to keep by accepting the `LispObject`.

Reply:
That alters the API. `(load-average t)` and `(load-average "asdf")` should return the same thing. It's not a pattern that fits in very well with the Rust compiler, which is what motivated me to try to use `Option` in https://github.com/Wilfred/remacs/pull/1126.

Reply:
@nickdrozd anything that is truthy will become true and nil is false. It should work as expected no?

Reply:
@nickdrozd in this case `use_floats` is not consumed by any of the functions inside `load_average`. In your PR it was.

Reply:
@carlosmn the thing to watch is where a function uses a value as a bool in itself but passes it on elsewhere and the original type is needed.

Something like this for instance:

    pub fn foo(value: LispObject) -> LispObject {
        if value.is_nil() {
            other_function(1)
        else {
            other_function(value)
        }
    }

In this case `other_function` is actually looking for a number and if we just converted the bool to a number it would be '1' which is not ok.

Reply:
`lisp_fn` is full of surprises. Just to assuage my lingering doubts though, could you add a test case that verifies that `(load-average)` returns a list of ints and `(load-average "asdf")` returns a list of floats? Or any other weird inputs (`(load-average '(list of floats))`, for instance).

Reply:
@nickdrozd perhaps we should dedicate some code to validating that `lisp_fn` behaves as expected. That way we 1) have less doubt and 2) have something to test any changes against.

Reply:
The tests look reasonable.